packages/cli/src/api/e2e.api.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Router } from 'express';
+import bodyParser from 'body-parser';
+import { v4 as uuid } from 'uuid';
+import config from '@/config';
+import * as Db from '@/Db';
+import { RoleNames, RoleScopes } from '@/databases/entities/Role';
+import { hashPassword } from '@/UserManagement/UserManagementHelper';
+
+if (process.env.E2E_TESTS !== 'true') {
+	console.error('E2E endpoints only allowed during E2E tests');
+	process.exit(1);
+}
+
+const tablesToTruncate = [
+	'shared_workflow',
+	'shared_credentials',
+	'webhook_entity',
+	'workflows_tags',
+	'credentials_entity',
+	'tag_entity',
+	'workflow_entity',
+	'execution_entity',
+	'settings',
+	'installed_packages',
+	'installed_nodes',
+	'user',
+	'role',
+];
+
+const truncateAll = async () => {
+	const { connection } = Db;
+	for (const table of tablesToTruncate) {
+		await connection.query(
+			`DELETE FROM ${table}; DELETE FROM sqlite_sequence WHERE name=${table};`,
+		);
+	}
+	config.set('userManagement.isInstanceOwnerSetUp', false);
+};
+
+const setupUserManagement = async () => {
+	const { connection } = Db;
+	await connection.query('INSERT INTO role (name, scope) VALUES ("owner", "global");');
+	const instanceOwnerRole = (await connection.query(
+		'SELECT last_insert_rowid() as insertId',
+	)) as Array<{ insertId: number }>;
+
+	const roles: Array<[RoleNames, RoleScopes]> = [
+		['member', 'global'],
+		['owner', 'workflow'],
+		['owner', 'credential'],
+		['user', 'credential'],
+		['editor', 'workflow'],
+	];
+
+	await Promise.all(
+		roles.map(async ([name, scope]) =>
+			connection.query(`INSERT INTO role (name, scope) VALUES ("${name}", "${scope}");`),
+		),
+	);
+	await connection.query(
+		`INSERT INTO user (id, globalRoleId) values ("${uuid()}", ${instanceOwnerRole[0].insertId})`,
+	);
+	await connection.query(
+		`INSERT INTO "settings" (key, value, loadOnStartup) values ('userManagement.isInstanceOwnerSetUp', 'false', true), ('userManagement.skipInstanceOwnerSetup', 'false', true)`,
+	);
+};
+
+export const e2eController = Router();
+
+e2eController.post('/db/reset', async (req, res) => {
+	await truncateAll();
+	await setupUserManagement();
```

Comment:
Should we run the migrations instead of using `setupUserManagement`? Faster to only run what we need, but at the cost of duplication.

Reply:
so, we `DROP` all the tables, and run the migration? I can try that.